game/dino_jump_main.py

```diff
@@ -0,0 +1,394 @@
+        for obs in self.curr_obs:
+            obs.x -= obs.SPEED
+
+    def choose_random_obstacle_attributes(self):
+        # Choose the next obstacle: cactus arrangement, low bird, or high bird
+        return np.random.choice(self.ALL_OBSTACLE_GROUPS), np.random.choice(
+            self.ALL_SPEEDS
+        )
+
+    def generate_obstacle(self):
+
+        # When the cooldown reaches zero, a new obstacle is generated
+        if self.curr_cooldown <= 0:
+            new_obstacle, new_speed = self.choose_random_obstacle_attributes()
+            for i, obs in enumerate(new_obstacle):
+                new_obs = SingleObstacle(obs, new_speed, self.WIDTH, self.AXES)
+                new_obs.x += (new_obs.WIDTH + 1) * i
+                self.curr_obs.append(new_obs)
+            self.curr_cooldown = self.BASE_COOLDOWN + random.choice(
+                self.COOLDOWN_VARIANCE
+            )
+
+        # Otherwise, we decrement the cooldown timer every loop
+        else:
+            self.curr_cooldown -= 1
+
+
+#############################################################################
+# DINO MODULE
+#############################################################################
+
+
+class Dino:
+
+    def __init__(self, GROUND_AXIS):
+
+        # Jumping Constants
+        self.JUMP_HEIGHT = 25  # specify how many pixels upwards the max-height goes
+        self.JUMP_TIME = 500  # specify time in milliseconds dino takes to land
+        self.UPDATE_WALK_FRAMES = 3  # how many frames to wait before dino switches legs
+
+        # Inherited from Game
+        self.GROUND_AXIS = GROUND_AXIS
+
+        # Get all pixel arrays for the dino from sprite import (dino:20x19)
+        self.JUMP_ARRAY = import_sprite("jump")
+        self.RRUN_ARRAY = import_sprite("right run")
+        self.LRUN_ARRAY = import_sprite("left run")
+        self.RDUCK_ARRAY = import_sprite("right duck")
+        self.LDUCK_ARRAY = import_sprite("left duck")
+        self.DONE_ARRAY = import_sprite("done")
+
+        # Define Array and Dimensions
+        self.ARRAY = self.JUMP_ARRAY  # the first array is still dino frame
+        self.HEIGHT, self.WIDTH = (
+            self.JUMP_ARRAY.shape
+        )  # get height and widrh to be able to call
+        self.OG_DINO_Y = (
+            self.GROUND_AXIS - self.HEIGHT
+        )  # define dino's top left pixel height
+        self.x = 5  # dino x stays put, how many pixels from left edge
+        self.y = self.OG_DINO_Y  # assigned to the y position
+        self.is_right_foot_down = (
+            True  # used to change between left and right foot arrays displayed
+        )
+        self.is_jumping = True  # detects if jumping signal is given
+        self.is_ducking = False  # detects if ducking signal is given
+        self.is_hit = False  # becomes true when obstacle and dino collide
+        self.jump_start_time = 0  # used to count time in jumping
+        self.curr_walk = (

```

Comment:
nit: To be more explicit maybe rename to curr_walk_frame

Reply:
Renamed!